### ISSUE TITLE
"TypeError: details.flatMap is not a function" when attempting "Delete All" of pools

### ISSUE THREAD

**Issue by philrz** [brimdata/brim#1815]
This issue was reported by a community user in a [public Slack thread](https://brimsec.slack.com/archives/C024JHN96SC/p1631055087018800). They were using the beta version based on Brim commit ec5da6b. In their own words:

> I don't know if I can recreate it or not but I tried to "Delete All" from the Brim App and got this error on the screen.  There was one pool left when I restarted the application and when I tried to delete it, it couldn't find it.  Tried to "Delete All" again and got the same thing but this time all pools where removed.  
>  ![image](https://user-images.githubusercontent.com/5934157/132425291-cf9adc7a-d37e-4c71-bfa9-be66ffd15a0c.png)
> 
> I think I had around 10 or so pools and the newest pcap I was looking at was around 400mb.

Since we didn't have a precise repro recipe, I tried importing lots of pcaps then trying "Delete All" in various permutations (e.g., varying which pool I had highlighted at the time I invoked it, which pools were open in which tabs, etc.) but I was unable to trigger the problem. Perhaps the UX team could look at the functions mentioned in the stack trace and provide some insight into what might trigger this.

**Comment by philrz** [comment]
The community user that originally reported the problem circled back with a solid repro recipe. The tl;dr is that the specific repro steps are likely unique to some breaking changes between beta releases, so there may not be exposure in the wider user base once we GA the next release. However, what we learn from investigating this specific repro could hopefully help us better handle this class of error.

To repro:

1. Launch Brim macOS beta release https://storage.googleapis.com/brimsec-releases/brim/v0.25.0-prerelease-ec5da6bb.0/macos/Brim-0.25.0-prerelease-ec5da6bb.0.dmg
2. Import a small test pcap such as [ifconfig.pcapng.gz](https://github.com/brimdata/brim/files/7138068/ifconfig.pcapng.gz) (after uncompressing). 
3. Quit the app and install/start the next Brim macOS beta release https://storage.googleapis.com/brimsec-releases/brim/v0.25.0-prerelease-7a72af02.0/macos/Brim-0.25.0-prerelease-7a72af02.0.dmg

Because of the changes to the Zed lake storage format between releases, the app will show a spinner. As shown in the attached video, if the user selects **Delete All** on the older pool at this point, the crash is triggered.

https://user-images.githubusercontent.com/5934157/132723262-1b679f66-376e-40bb-a533-449ec0e61665.mp4

Full disclosure: I'm basically responsible for walking people into this issue. With the recent beta releases that have changed Zed storage format, I did post notices providing tips to export data from the old app version before installing/launching the new version and re-importing, and those tips included deleting Brim's `data` directory before installing the newer version, which would have removed the old pools, prevented the app from coming up in this "spinning" state, and there'd have been no pools to **Delete All** and trigger the issue. My mistake was that I forgot to advise users that _don't_ want to keep their data that they'd _still_ need to delete Brim's `data` directory. I've now adjusted my beta posting in hopes users will avoid this hazard for the most recent beta release.

**Comment by philrz** [comment]
Verified in Brim commit 7d1e6b1d using artifact https://storage.googleapis.com/brimsec-releases/brim/v0.26.0-prerelease-7d1e6b1d.0/macos/Brim-0.26.0-prerelease-7d1e6b1d.0.dmg.

To prepare, I went back and did another repro by starting with the first two repro steps in the most recent comment above, but then in the third step I installed the GA Brim tagged `v0.25.0` and confirmed I still saw the crash screen. Then for the verification, I went back and repeated the same two first repro steps, but then in the third step I installed Brim artifact https://storage.googleapis.com/brimsec-releases/brim/v0.26.0-prerelease-7d1e6b1d.0/macos/Brim-0.26.0-prerelease-7d1e6b1d.0.dmg that includes the fix in the linked PR.

As shown in the attached video, now instead of a crash screen the user gets a much better error message.

https://user-images.githubusercontent.com/5934157/134590412-43a92062-819c-4269-af42-6bc9008849ad.mp4

Now, for this specific repro, there's still a spinner, the "Delete All" ultimately fails, etc., but that's orthogonal to the issue at hand. Per the notes above, this issue was only triggered because of the absence of migration code to preserve pools between beta releases, and that code intentionally doesn't exist because the guidance for beta testers was that they'd need to start fresh by deleting their pools directory from outside the app. But the repro still proves the point that the error message handling is improved.

Thanks @mason-fish!
